src/coreComponents/codingUtilities/StringUtilities.hpp

```diff
@@ -140,6 +140,38 @@ array1d< T > fromStringToArray( string const & str )
   return v;
 }
 
+/**
+ * @brief Take a numerical value and convert/scale it to a string with a metric
+ *  prefix. i.e. Kilo, Mega, Giga, Tera, Peta, Exa
+ *
+ * @tparam T Type of the value to be converted
+ * @param value The value to be converted
+ * @return String containging the scaled value.
+ */
+template< typename T >
+string toMetricPrefixString( T const & value )
+{
+  char const prefixes[7] = { ' ', 'K', 'M', 'G', 'T', 'P', 'E'};
+
+  T rval;
+  string retString;
+  for( int a=6; a>=0; --a )
+  {
+    T const scaleFactor = pow( 10.0, 3*a );
+    if( value > scaleFactor || a==0 )
+    {
+      rval = value / scaleFactor;
+      rval = round( rval * 10 ) / 10;
+      retString = std::to_string( rval );
+      int const preDecimalDigits = retString.find( "." );
+      int const digits = (preDecimalDigits<=2) ? preDecimalDigits+2 : preDecimalDigits;
+      retString = retString.substr( 0, digits ) + " " + prefixes[a];
+      break;
+    }
+  }
+  return retString;
+}
```

Comment:
I did not go though all the cases, but I could find online something like
```
    vector<string> const units{ "B", "kB", "MB", "GB", "TB", "PB", "EB" };
    int exponent = std::log( value ) / std::log( 1000. );
    fmt::print("{:.1f} {}", value / std::pow( 1000., exponent ), units[exponent]);
```
Maybe there's better to do than the `std::pow` division anyway?
This uses the lib `fmt`, but we have it.

Reply:
I don't see why this would be better. Also, I would like the digits to the right of the decimal only in the case of single digits to the left of the decimal (i.e. 2-3 digits of precision). Can you illuminate me on why this suggestion is preferable?

Reply:
Formatting is not such an easy task. So if we can delegate this task to a TPL (that will eventually be part of the `std`), it saves us some work.

I do not know if `fmt` does what you want out of the box, but if not, we can simply
- don't do it;
- do something like `string const format = value / std::pow( 1000., exponent ) >= 100 ? "{:.0f} {}": "{:.1f} {}";` and it should work as expected.

Saving us some code is always nice I think.